src/Foundation/Features/CatalogContent/Extensions.cs

```diff
@@ -57,8 +67,12 @@ public static ProductTileViewModel GetProductTileViewModel(this EntryContentBase
 
             if (entry is GenericProduct)
             {
-                entryUrl = UrlResolver.Value.GetUrl(product.ContentLink);
-                firstCode = isDiscounted ? ReferenceConverter.Value.GetCode(minDiscountPrice.Key) : minPrice.CatalogEntryCode;
+                var variants = GetProductVariants(entry);
+                if (variants != null && variants.Any())
+                {
+                    firstCode = variants.First().Code;
+                }
+                entryUrl = UrlResolver.Value.GetUrl(entry.ContentLink);
```

Comment:
Because the showed price in the view is the min discounted price, so I think getting the first code by this way may be incorrect.

Reply:
There are two prices for the viewmodel: PlacedPrice and DiscountedPrice. The PlacedPrice is the price without a discount. The DiscountedPrice is the price after discounts are applied by the promotion engine to the PlacedPrice. Visually, this means that the PlacedPrice should always show and, if there is a discount on the price, the PlacedPrice should have a line through it and the DiscountedPrice should be shown.

The first code is retrieved because products do not contain prices - SKUs/Variants do. So the code retrieves the first SKU's code to find a price for a product.

Reply:
> There are two prices for the viewmodel: PlacedPrice and DiscountedPrice. The PlacedPrice is the price without a discount. The DiscountedPrice is the price after discounts are applied by the promotion engine to the PlacedPrice. Visually, this means that the PlacedPrice should always show and, if there is a discount on the price, the PlacedPrice should have a line through it and the DiscountedPrice should be shown.
> 
> The first code is retrieved because products do not contain prices - SKUs/Variants do. So the code retrieves the first SKU's code to find a price for a product.

For example:
Product A link to variants (A1, A2, A3) and  the price of variants are: 
1. A1: 3
2. A2: 2
3. A3: 1

When we get the display price of product, A3 will be gotten. But  the first code is A1.

Reply:
So your concern is that the First() method on the variants list will return A3? I'm not sure I see the problem. The call to First() is simply to find a SKU code that a price can be returned for. What do you propose alternatively?

Reply:
You can keep the latest code that used to get the exact variant code. I mean 
` firstCode = isDiscounted ? ReferenceConverter.Value.GetCode(minDiscountPrice.Key) : minPrice.CatalogEntryCode;`

Reply:
This code that you are referring to is the source of very difficult to debug price inconsistencies. The price returned from this method can be different from the price seen on a category search page and in checkout. I rewrote the GetProductTileViewModel() method to be consistent with the way prices and discounted prices are calculated on the category search page and in checkout **and** much easier to debug. To keep the line of code you refer to here means keeping the method the same, producing those inconsistent results.

Reply:
Because `firstCode = variants.First().Code;` is the root code. I remember that the product is getting the min prices of related variants and show on the frontend web. So, if you get the first code, it may be incorrect when the user add to cart in the quick view. After merging this pull request, if you have any other bugs relate to this, please add issues to dashboard  and add the description to that issue. Thank you.

Reply:
@shannongrayepiserver Please sign your CLA, so I can merge the PR. And in the next times, if you want fix something or create a PR, please add a issues and clearly description, so we will be able to better understand and easier to review the code. Thanks.

Reply:
Hi - I clicked on the CLA link and it says I've signed it. I'm a US-based Episerver/Optimizely Solution Architect...

Reply:
Regarding the variants.First(): When a product teaser is shown for a product (which is what this code is about), it should show the same price as shown on the product detail page. Otherwise, it would be confusing and you might feel tricked (and its not helpful in a demo).
If you look at the CatalogEntryViewModelFactory Create() method, you can see on line 107 that its doing the same thing, getting the first variant in the list of variants (FirstOrDefault()). There is some additional code for filtering SKUs based on availability in a market in the GatalogEntryViewModelFactory class if you want it to be more correct.

Reply:
@shannongrayepiserver Not sure why your CLA status is still pending.
![image](https://user-images.githubusercontent.com/56108286/112104319-a22fb180-8bdd-11eb-953b-06ebb0a7c42a.png)

Reply:
Not sure what happened here but this commit is now merged into develop